src/Neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -54,10 +54,18 @@ public void AddTransaction(Transaction tx)
         /// <param name="tx">The specified <see cref="Transaction"/>.</param>
         /// <param name="conflictingTxs">The list of <see cref="Transaction"/> that conflicts with the specified one and are to be removed from the pool.</param>
         /// <param name="snapshot">The snapshot used to verify the <see cref="Transaction"/>.</param>
+        /// <param name="settings">The <see cref="ProtocolSettings"/> used to verify the <see cref="Transaction"/>.</param>
         /// <returns><see langword="true"/> if the <see cref="Transaction"/> passes the check; otherwise, <see langword="false"/>.</returns>
-        public bool CheckTransaction(Transaction tx, IEnumerable<Transaction> conflictingTxs, DataCache snapshot)
+        public bool CheckTransaction(Transaction tx, IEnumerable<Transaction> conflictingTxs, DataCache snapshot, ProtocolSettings settings)
         {
-            var balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
+            BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
+            var currentIndex = NativeContract.Ledger.CurrentIndex(snapshot);
+            var nextIndex = currentIndex + 1;
+            if (settings.IsHardforkEnabled(Hardfork.HF_Faun, nextIndex) &&
+                !NativeContract.ContractManagement.IsContract(snapshot, tx.Sender))
+            {
+                balance += NativeContract.NEO.UnclaimedGas(snapshot, tx.Sender, nextIndex);
```

Comment:
This has cost for every transaction, will affect TPS metrics.

Reply:
Same as in https://github.com/neo-project/neo/issues/1078#issuecomment-1811187629, different real cases can be covered in different ways. To me the problem is not worth the added overhead and added inconsistency as well (contract addresses are special now wrt this, no good).